### ISSUE TITLE
SPInstallLanguagePack install not working if Config_Database does not exist

### ISSUE THREAD

**Issue by luigilink** [PowerShell/SharePointDsc#410]
**Details of the scenario you try and problem that is occurring:**
Install Language Pack after installation of SharePoint binaries.
This type of scenario is current on a new sharepoint farm:

1. Install of SharePoint Binaries
2. Install of SharePoint Language Pack(s)
3. Install of Cumulative Updates
4. Create SPFarm 

**The DSC configuration that is using the resource:**
```
SPInstallLanguagePack APPLICATION_SpsLangPackFr
{
         DependsOn = '[SPInstall]APPLICATION_SpsInstallSharePoint'
         Ensure = 'Present'
         BinaryDir = "E:\INSTALL\SPS\LP"
         PsDscRunAsCredential = $InstallAccount
}
```

**Version of the Operating System and PowerShell the DSC Target Node is running:**

- PowerShell Version: 5.0.10586.117
- OS Version : Windows 2012 R2

**Version of the DSC module you're using:** SharePointDSC 1.3.0.0

**Get-DscConfiguration Verbose ERROR:**
```
Get-DscConfiguration : La ressource DSC PowerShell MSFT_SPInstallLanguagePack n'a pas pu exécuter la fonctionnalité
Get-TargetResource. Message d'erreur: Cannot access the local farm. Verify that the local farm is properly configured,
currently available, and that you have the appropriate permissions to access the database before trying again.
Au caractère Ligne:1 : 1
+ Get-DscConfiguration -Verbose
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (MSFT_DSCLocalConfigurationManager:root/Microsoft/...gurationManager)
    [Get-DscConfiguration], CimException
    + FullyQualifiedErrorId : ProviderOperationExecutionFailure,Get-DscConfiguration
```

**Comment by BrianFarnhill** [comment]
@ykuijs Can you have a look at the language pack one to see if we are depending on a farm existing? From memory I think we actually might be on that one.

**Comment by luigilink** [comment]
Fix contribution in pull request #413

**Comment by ykuijs** [comment]
Correct, the resource currently checks if the farm exists, since it uses the product information in the ConfigDB. Will review the suggested fix.

**Comment by ykuijs** [comment]
I never though of this scenario when designing these resources. The suggested fix in PR413 will fix the issue with the installation of Language Pack, but the resource to install CU's depends on retrieving the version information from the Config DB. The above mentioned order won't work with these resources. That has to be:
1. Install of SharePoint Binaries
2. Install of SharePoint Language Pack(s)
3. Create SPFarm 
4. Install of Cumulative Updates

**Comment by BrianFarnhill** [comment]
OK I think we need to try to determine a way to get the CU's to be able to be installed before we create the farm - I know slipstreaming covers this (and I'm happy to let #413 go through as long as we capture this in the documentation of the CU resource) but I think we need to add a separate issue to address this. I feel like we can do something like look at the version number of Microsoft.SharePoint.dll (I'm sure we used to do this at some point, maybe? My memory is a little hazy on it) because that will always be installed at the same location (we can wildcard the hive number to catch 15 or 16) and it will be there as soon as the binaries are installed. That should cover us for that scenario shouldn't it @ykuijs ? 

I'll merge #413 in and we can add separate PR's issues to capture the rest of this after we resolve this discussion.

**Comment by ykuijs** [comment]
Slipstreaming CUs is not the best way if you are also using Language Packs. Since CUs are multi lingual, they need to be installed last, in order to update all installed language packs. If you slipstream it into the SP install and then install language packs, your SharePoint install will be on the CU patch level, but your language packs won't.

The only way to determine the version of all installed components is to use the information from the ConfigDB. Checking the Microsoft.SharePoint.dll is just one of the components that make up SharePoint. If you installed a security patch that just updates this dll, your code thinks it is running a certain version of SharePoint, where actually it is just that one file which is on that version.

The product status page shows all components with their corresponding versions. According to the patching article on Stefan Gossner's blog: "To get a more reliable picture you really have to look at the Patch Status page ("Check product and patch installation status"). It contains the patch level for each installed component."

https://blogs.technet.microsoft.com/stefan_gossner/2014/08/18/sharepoint-patching-demystified/

**Comment by BrianFarnhill** [comment]
@ykuijs OK happy with that (I live in an English speaking country so haven't had to do language packs for production environments before, so appreciate the perspective).

Happy to go through and update our doco to make the process clear on that then just so that when people like me come to it we can get it right :)

**Comment by ykuijs** [comment]
I hate language packs, makes troubleshooting harder (localized error messages) and I am always searching for the link I am looking for (translated text in for example Site Settings, also with a different order than in English). So try to stay as far away from it as possible :-) But I do understand they are used a lot in some countries ;-)

Agree that we have to update the docu. How to proceed on this? What docu needs updating? Who will be doing that?

**Comment by BrianFarnhill** [comment]
Just opened a new issue for the doo update - should be a quick one to do. I'm closing this one off now though as the main issue here is done and fix merged in #413. Thanks guys.
